Fixes for Moodle 4.1 compatibility

marxjohnson opened the pull request:
Resolves #152

brendanheywood commented on cli/lib.php:
```diff
@@ -98,7 +98,7 @@ function safety_checks($dryrun) {
 
     if ($DB->count_records_sql($csql, $params)) {
-        $namefields = "u." . implode(', u.', get_all_user_name_fields());
+        $namefields = "u." . implode(', u.', \core_user\fields::get_name_fields());
```
won't this break in older moodles where this doesn't exist yet? Either we need to detect the version and call the old function of pick a version and cut a new stable branch. It would be better have this in a unit test to expose that issue across versions too

marxjohnson replied:
This function was introduced on Moodle 3.11, so you're right that it won't work on Moodle 3.10 which this branch supports (although it's no longer a supported release). Personally I prefer separate stable branches over version-detection code, as the latter feels like technical debt. Shall I create a new branch and change this PR's target?

brendanheywood replied:
Yup please cut a new stable

marxjohnson replied:
Actually, I think you'll have to do that for me. I don't have write access and GitHub doesn't have a way for me to submit a new branch from my repo to this one. I think it would make sense to call it MOODLE_311_STABLE as this and the removed user fields both apply from 3.11 onwards.

brendanheywood replied:
I've made the new branch as a clone of the 310 one so you can make new prs against that now thanks

marxjohnson commented:
Hi @brendanheywood ,
I've rebased this on the new stable branch, bumped the version and supported core versions, and added the new branch to the README. I think this is ready to merge now.

marxjohnson commented:
@brendanheywood There we go, all green now on 3.11-4.1.

danmarsden commented:
looks good to me - I've also updated the default branch here it github to "MOODLE_311_STABLE" - I didn't delete the old "master" branch as it will break AU pipelines using the older master branch.
